rfd/0207-encrypted-session-recordings.md

```diff
@@ -0,0 +1,293 @@
+---
+authors: Alan Parra (alan.parra@goteleport.com), Erik Tate (erik.tate@goteleport.com) 
+state: draft
+---
+
+# RFD 207 - Encrypted Session Recordings
+
+## Required Approvers
+* Engineering: @rosstimothy, @zmb3, @espadolini, @nklaassen
+* Security: doyensec
+
+## What
+
+This document proposes an approach to encrypting session recording data before
+writing to disk or any S3 compatible long term storage.
+
+## Why
+
+File recordings on a node can be easily tampered with by users with enough access.
+This could even occur within the session being recorded if the host user has root
+access.
+
+## Details
+
+This document should fulfill the following requirements:
+- Ability to encrypt session recording data during intermediate disk writes and long term storage.
+- Ability to provide an externally managed public key for encryption.
```

Comment:
is this a direct customer ask? I'm wondering why we don't want to use a Teleport-managed key so that sessions could be decrypted on Auth and played as normal in the UI and from tsh. Or if that could be an option, with the externally-managed key being opt-in only for customers who explicitly want that

Reply:
I dropped this entirely for now. We can always reintroduce this if we have an actual need for it in the future